docs/design-archive/runtime-manager-design-doc.md

```diff
@@ -0,0 +1,116 @@
+# Design of RuntimeManager
+
+* [Design of RuntimeManager](#summary)
+    * [Summary](#summary)
+    * [Goals](#goals)
+    * [Components](#components)
+        * [RuntimeManager](#runtimemanager)
+        * [RuntimePlugins](#runtimeplugins)
+    * [Architecture of RuntimeManager](#architecture)
+        * [CRI Server](#cri-server)
+        * [Plugins Manager](#plugins-manager)
+        * [Runtime Dispatcher](#runtime-dispatcher)
+        * [Store](#store)
+    * [RuntimePlugins](#examples-for-hooks-extensions)
+        * [How to Register Plugins](#how-to-register-plugins)
+        * [Protocols between RuntimeManager and Plugins](#protocols-between-runtimemanager-and-plugins)
+        * [Examples for Runtime Plugins](#examples-for-runtime-plugins)
+    * [Installation](#installation)
+
+## Summary
+
+RuntimeManager acts as a proxy between kubelet and containerd(dockerd under dockershim scenario), which is designed to
+intercept CRI request, and apply some resource management policies, such as setting different cgroup parameters by pod
+priorities under hybrid scenario, applying new isolation policies for latest Linux kernel, CPU architecture, and etc.
+
+There are two components involved, RuntimeManager and RuntimePlugins.
+
+![image](../images/runtime-manager-architecture.svg)
+
+## Goals
+
+- Enhance resource management for scenario like hybridization.
```

Comment:
Enhance resource management for QoS based Scheduling

Reply:
> Enhance resource management for QoS based Scheduling

fixed